src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -96,19 +135,19 @@ const bundleCustomElements = async (
           buildCtx.diagnostics.push(...optimizeResults.diagnostics);
           if (!hasError(optimizeResults.diagnostics) && typeof optimizeResults.output === 'string') {
             code = optimizeResults.output;
-            sourceMap = optimizeResults.sourceMap;
           }
-          if (sourceMap) {
+          if (optimizeResults.sourceMap) {
+            sourceMap = optimizeResults.sourceMap;
             code = code + getSourceMappingUrlForEndOfFile(bundle.fileName);
             await compilerCtx.fs.writeFile(
-              join(outputTarget.dir, bundle.fileName + '.map'),
+              join(outputTarget.dir!, bundle.fileName + '.map'),
               JSON.stringify(sourceMap),
               {
                 outputTargetType: outputTarget.type,
               }
             );
           }
-          await compilerCtx.fs.writeFile(join(outputTarget.dir, bundle.fileName), code, {
+          await compilerCtx.fs.writeFile(join(outputTarget.dir!, bundle.fileName), code, {
```

Comment:
that makes sense to me

Reply:
I do strongly believe that `dir` _should_ never be null because `validateCustomElement` _should_ always have been called previously and that does ensure that `dir` is always set. However, we don't have anything in the type system enforcing that contract right now, since the `OutputTargetDistCustomElements` has an optional `dir` property on it and we don't do any looser type -> more restrictive type transformation when we do the configuration validation process (yet, if I have my way haha).

Anywhoo - because of that and the possibility (in general) for the contract between this bit of code and the rest of the codebase to change I went ahead and declared an `outputTargetDir` variable which pulls out the value and then added a diagnostic error if it isn't a string, so that in the future if something changes and this code is called without calling `validateCustomElement` we'll get an error.